lua/lint/parser.lua

```diff
@@ -11,6 +11,75 @@ local severity_by_qftype = {
   N = vd.severity.HINT,
 }
 
+---@class lint.SarifOptions
+---@field default_end_col "+1" | "eol" the default end column, "+1" (defaults to the end of the line)
+
+---Return a parse function for the Static Analysis Results Interchange Format (SARIF).
+---https://sarifweb.azurewebsites.net/
+---@param skeleton? table<string, any> | vim.Diagnostic default values
+---@param opts? lint.SarifOptions SARIF-related options
+---@return fun(output: string, bufnr: number): vim.Diagnostic[] parser a SARIF parser
+function M.for_sarif(skeleton, opts)
+  skeleton = skeleton or {}
+  skeleton.severity = skeleton.severity or vd.severity.ERROR
+
+  opts = opts or {}
+  opts.default_end_col = opts.default_end_col or "eol"
+  local default_end_col = opts.default_end_col == "eol" and 999999 or nil
+
+  local severities = {
+    error = vd.severity.ERROR,
+    warning = vd.severity.WARN,
+    note = vd.severity.INFO,
+  }
+
+  ---@param output string the output of the tool
+  ---@param linter_bufnr number the number of the buffer the linter ran on
+  ---@return vim.Diagnostic[] the diagnostics
+  return function(output, linter_bufnr)
+    local diagnostics = {}
+
+    local decoded = vim.json.decode(output) or {}
+
+    local run = decoded.runs and decoded.runs[1] or {}
```

Comment:
I'm not very familiar with sarif but based on the [runs spec](https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/sarif-v2.1.0-errata01-os-complete.html#_Toc141790732) I get the impression that it might be more appropriate to loop over all runs and collect diagnostics for all of them?